app/js/enhancedDate.js

```diff
@@ -1,5 +1,99 @@
-/* 
+/*
  * add your module definition for enhancedDate here
  * assign the module to a global variable
- * /
+ */
 
+var enhancedDate = (function iffe() {
+  'use strict';
+  function now() {
+    return new Date();
+  }
+
+  var dateObjects = {
+    dateTime: now(),
+    monthNames: ['January',
+      'February',
+      'March',
+      'April',
+      'May',
+      'June',
+      'July',
+      'August',
+      'September',
+      'October',
+      'November',
+      'December'],
+    dayNames: ['Sunday',
+      'Monday',
+      'Tuesday',
+      'Wednesday',
+      'Thursday',
+      'Friday',
+      'Saturday']
+  };
+
+  return {
+    setDate: function setDate(e) {
+      if (e instanceof Date) {
+        dateObjects.dateTime = e;
+      }
+      else if (!isNaN(e)) {
+        dateObjects.dateTime = new Date(e);
+      }
+      else if (e === undefined) {
+        dateObjects.dateTime = new Date();
+      }
+      else {
+        console.warn('argument is not instanceof Date');
+      }
+      return dateObjects.dateTime;
+    },
+    getDate: function getDate(e) {
+      if (e === true) {
+        return dateObjects.dateTime;
+      }
+    },
+    getDayName: function getDayName() {
+      var day = dateObjects.dateTime.getDay();
+      return dateObjects.dayNames[day];
+    },
+    getMonthName: function getMonthName() {
+      var month = dateObjects.dateTime.getMonth();
+      return dateObjects.monthNames[month];
+    },
+    isFuture: function isFuture() {
+      if (now().getTime() < dateObjects.dateTime.getTime()) {
+        return true;
+      }
+      else {
+        return false;
+      }
+    },
+    isToday: function isToday() {
+      var year = now().getFullYear();
+      var month = now().getMonth();
+      var day = now().getDate();
+      var nextDay = new Date(year, month, day).getTime() + 86400000;
```

Comment:
Sort of. What I mean is declare a variable as a constant somewhere and use
that variable there - makes the code more self-documenting:

var MILLISECONDS_IN_DAY = 86400000;

var nextDay = new Date(year, month, day).getTime() + MILLISECONDS_IN_DAY;

In general, avoid magic numbers in code so the code's intent is immediately
obvious.

On Sun, Apr 12, 2015 at 8:29 AM, httpstergeek <notifications@github.com>
wrote:

> In app/js/enhancedDate.js
> <https://github.com/JSCRIPT300-Spring2015/Module_2/pull/1#discussion_r28204376>
> :
>
> > +      var month = dateObjects.dateTime.getMonth();
> > +      return dateObjects.monthNames[month];
> > +    },
> > +    isFuture: function isFuture() {
> > +      if (now().getTime() < dateObjects.dateTime.getTime()) {
> > +        return true;
> > +      }
> > +      else {
> > +        return false;
> > +      }
> > +    },
> > +    isToday: function isToday() {
> > +      var year = now().getFullYear();
> > +      var month = now().getMonth();
> > +      var day = now().getDate();
> > +      var nextDay = new Date(year, month, day).getTime() + 86400000;
>
> just for clarification. Are you saying that line 76 the value of day in
> milliseconds should be stored else where?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JSCRIPT300-Spring2015/Module_2/pull/1/files#r28204376>
> .
>